### PR TITLE
Use type casting instead of *val() method

### PR DESCRIPTION
**Pull request by MorrisJobke**
It should be up to 6x faster

### CODE REVIEW

**Review by blizzz — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Comment by blizzz** [comment]
@MorrisJobke there's a test failing

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Conflicting files
apps/user_ldap/lib/Helper.php

**Comment by nickvergessen** [comment]
> There was 1 failure:
> 
> 1) Test\User\DatabaseTest::testSearch
> Failed asserting that 3 is identical to 2.
> 
> /drone/src/github.com/nextcloud/server/tests/lib/User/Backend.php:112

CI drunk, or related?

File: apps/user_ldap/lib/Controller/ConfigAPIController.php [hunk 1]
```diff
@@ -285,7 +285,7 @@ public function show($configID, $showPassword = false) {
 
 			$config = new Configuration($configID);
 			$data = $config->getConfiguration();
-			if(!boolval(intval($showPassword))) {
+			if(!(bool)((int)$showPassword)) {
```
Comment (ChristophWurst) [inline apps/user_ldap/lib/Controller/ConfigAPIController.php]: 🙈

**Review by ChristophWurst — APPROVED** [review]

File: apps/user_ldap/lib/Controller/ConfigAPIController.php (hunk 1 above)
Comment (blizzz) [reply apps/user_ldap/lib/Controller/ConfigAPIController.php]: Yeah, you can omit the bool-casting.

Comment (MorrisJobke) [reply apps/user_ldap/lib/Controller/ConfigAPIController.php]: I also noticed this, but thought: okay let's keep it like that for now.

**Comment by MorrisJobke** [comment]
Fixed the conflict and fixed the double casting.

**Pull request merged**